src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,180 +1,165 @@
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
 import { RoomModal } from "components/templates/PartyMap/components";
 
-import "./NavSearchBar.scss";
 import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+import { NavSearchResult } from "./NavSearchResult";
+
+import "./NavSearchBar.scss";
+
+const emptyEventsArray: VenueEvent[] = [];
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
   const onSearchInputChange = useCallback(
-    (e) => setSearchQuery(e.target.value),
+    (e: ChangeEvent<HTMLInputElement>) =>
+      setSearchQuery(e.target.value.toLowerCase()),
     []
   );
 
-  const [searchResult, setSearchResult] = useState<SearchResult>({
-    rooms: [],
-    users: [],
-    events: [],
-  });
+  const clearSearchQuery = useCallback(() => {
+    setSearchQuery("");
+  }, []);
 
   const [selectedRoom, setSelectedRoom] = useState<Room>();
   const hasSelectedRoom = !!selectedRoom;
 
   const venue = useSelector(currentVenueSelectorData);
-  const venueEvents = useSelector(venueEventsSelector) ?? [];
+  const venueEvents = useSelector(venueEventsSelector) ?? emptyEventsArray;
   const { worldUsers } = useWorldUsers();
 
   const { openUserProfileModal } = useProfileModalControls();
 
-  useEffect(() => {
-    const normalizedSearchQuery = searchQuery.toLowerCase();
-    if (!normalizedSearchQuery) {
-      setSearchResult({
-        rooms: [],
-        users: [],
-        events: [],
-      });
-      return;
+  const foundRooms = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
     }
 
-    const venueUsersResults = worldUsers.filter((user) =>
-      user.partyName?.toLowerCase()?.includes(normalizedSearchQuery)
+    /* @debt we really shouldn't be using the index as part of the key here, it's unstable.. but rooms don't have a unique identifier */
+    return (
+      venue?.rooms
+        ?.filter(
+          (room) =>
+            room.title.toLowerCase().includes(searchQuery) &&
+            room.type !== RoomTypes.unclickable
+        )
+        .map((room, index) => (
+          <NavSearchResult
+            key={`room-${room.title}-${index}`}
+            title={room.title}
+            description="Room"
+            image={room.image_url}
+            onClick={() => {
+              setSelectedRoom(room);
+              clearSearchQuery();
+            }}
+          />
+        )) ?? []
     );
+  }, [searchQuery, venue, clearSearchQuery]);
 
-    const venueEventsResults = venueEvents.filter((event) =>
-      event.name.toLowerCase().includes(normalizedSearchQuery)
-    );
+  const foundUsers = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
```

Comment:
[nit] We can do this in 1 line for guards like this.

```suggestion
    if (!searchQuery) return [];
```

Reply:
fixed: https://github.com/sparkletown/sparkle/pull/1328/commits/0971387bcae59c871d82280587adfb3a590f5ac9